File: ruby-gem/lib/models/route.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #Route
require_relative '../core/semantic_object'
require_relative 'where_subject'


module DfcLinkmlConnector
  module Models
    class Route < WhereSubject
      SEMANTIC_TYPE = "dfc-b:Route".freeze

      # @return [String]
      attr_accessor :step

      # @return [String]
      attr_accessor :use_vehicle

      # @return [Feature]
      attr_accessor :geo_json_feature

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, step: nil, useVehicle: nil, geoJsonFeature: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, step: nil, useVehicle: nil, geoJsonFeature: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @step = step
        @use_vehicle = useVehicle
        @geo_json_feature = geoJsonFeature
        self.semanticType = "dfc-b:Route"
        registerSemanticProperty("dfc-b:Route:has_step", &method("step")).valueSetter = method("step=")
        registerSemanticProperty("dfc-b:Route:use_vehicle", &method("use_vehicle")).valueSetter = method("use_vehicle=")
        registerSemanticProperty("dfc-b:Route:has_geo_json_feature", &method("geo_json_feature")).valueSetter = method("geo_json_feature=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end

File: typescript-connector/src/models/Route.ts
// Class from DFC Business Ontology: #Route
import { SemanticObject } from "../core/SemanticObject.js";
import { WhereSubject, type WhereSubjectParams } from "./WhereSubject.js";
import type { Feature } from "./Feature.js";

export interface RouteParams extends WhereSubjectParams {
  hasStep?: string;
  useVehicle?: string;
  hasGeoJsonFeature?: Feature;
}
export class Route extends WhereSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:Route";
  }

  hasStep?: string;
  useVehicle?: string;
  hasGeoJsonFeature?: Feature;

  constructor(
    semanticId: string,
    params?: RouteParams,
  ) {
    super(semanticId, params);
    this.hasStep = params?.hasStep;
    this.useVehicle = params?.useVehicle;
    this.hasGeoJsonFeature = params?.hasGeoJsonFeature;
    this.semanticType = Route.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:Route:has_step", () => this.hasStep);
    this.registerSemanticProperty("dfc-b:Route:use_vehicle", () => this.useVehicle);
    this.registerSemanticProperty("dfc-b:Route:has_geo_json_feature", () => this.hasGeoJsonFeature);
  }
  static {
    SemanticObject.typeRegistry.set(Route.SEMANTIC_TYPE, Route);
  }
}
